CommonTools/PileupAlgos/src/PuppiAlgo.cc

```diff
@@ -140,22 +167,32 @@ void PuppiAlgo::computeMedRMS(const unsigned int &iAlgo,const double &iPVFrac) {
     // some ways to do corrections to fRMS and fMedian
     fRMS [iAlgo] *= fRMSScaleFactor[iAlgo];
 
-    fRMS[iAlgo]    *= fRMSEtaSF;
-    fMedian[iAlgo] *= fMedEtaSF;
-    
-    if(!fAdjust[iAlgo]) return;
-    //Adjust the p-value to correspond to the median
-    std::sort(fPupsPV.begin(),fPupsPV.end());
-    int lNPV = 0; 
-    for(unsigned int i0 = 0; i0 < fPupsPV.size(); i0++) if(fPupsPV[i0] <= lMed ) lNPV++;
-    double lAdjust = double(lNPV)/double(fPupsPV.size()+fNCount[iAlgo]);
-    if(lAdjust > 0) fMedian[iAlgo] -= sqrt(ROOT::Math::chisquared_quantile(lAdjust,1.)*fRMS[iAlgo]);
+    if(fAdjust[iAlgo]){ 
+        //Adjust the p-value to correspond to the median
+        std::sort(fPupsPV.begin(),fPupsPV.end());
+        int lNPV = 0; 
+        for(unsigned int i0 = 0; i0 < fPupsPV.size(); i0++) if(fPupsPV[i0] <= lMed ) lNPV++;
+        double lAdjust = double(lNPV)/double(lNPV+0.5*fNCount[iAlgo]);
+        if(lAdjust > 0) {
+          fMedian[iAlgo] -= sqrt(ROOT::Math::chisquared_quantile(lAdjust,1.)*fRMS[iAlgo]);
+          fRMS[iAlgo]    -= sqrt(ROOT::Math::chisquared_quantile(lAdjust,1.)*fRMS[iAlgo]);
```

Comment:
it looks a bit odd that both the median and RMS are shifted by the same value.
Is this done correctly?
Also, copy-paste of the same multi-operand code is not good (use a temporary).